zerver/lib/actions.py

```diff
@@ -6704,8 +6704,12 @@ def do_unmute_user(mute_object: MutedUser) -> None:
 
 def do_mark_hotspot_as_read(user: UserProfile, hotspot: str) -> None:
     UserHotspot.objects.get_or_create(user=user, hotspot=hotspot)
-    event = dict(type="hotspots", hotspots=get_next_hotspots(user))
-    send_event(user.realm, event, [user.id])
+
+    # Avoid sending the next hotspots if it's the non-intro hotspot
+    # because all the required hotspots are already sent.
+    if hotspot not in NON_INTRO_HOTSPOTS:
+        event = dict(type="hotspots", hotspots=get_next_hotspots(user))
+        send_event(user.realm, event, [user.id])
```

Comment:
More correct would be to send the data unconditionally, and have the frontend avoid double-rendering things if it gets a hotspot twice (which is effectively a latent frontend bug).  The structure of this API is for the backend to tell the frontend which hotspots should be currently enabled, otherwise we'd be doing some diffing algorithm on the backend.